modules/programs/powerline-go.nix

```diff
@@ -0,0 +1,271 @@
+    }
+    {
+      name = "colorizeHostname";
+      description =
+        "Colorize the hostname based on a hash of itself, or use the PLGO_HOSTNAMEFG and/or PLGO_HOSTNAMEBG env vars.";
+      type = types.bool;
+    }
+    {
+      name = "condensed";
+      description = "Remove spacing between segments";
+      type = types.bool;
+    }
+    {
+      name = "cwdMaxDepth";
+      description = "Maximum number of directories to show in path (default 5)";
+      type = types.int;
+    }
+    {
+      name = "cwdMaxDirSize";
+      description =
+        "Maximum number of letters displayed for each directory in the path (default -1)";
+      type = types.int;
+    }
+    {
+      name = "cwdMode";
+      description = ''How to display the current directory (default "fancy")'';
+      type = types.enum [ "fancy" "plain" "dironly" ];
+    }
+    {
+      name = "eastAsianWidth";
+      description = "Use East Asian Ambiguous Widths";
+      type = types.bool;
+    }
+    {
+      name = "gitAssumeUnchangedSize";
+      description =
+        "Disable checking for changed/edited files in git repositories where the index is larger than this size (in KB), improves performance (default 2048)";
+      type = types.int;
+    }
+    {
+      name = "hostnameOnlyIfSsh";
+      description = "Show hostname only for SSH connections";
+      type = types.bool;
+    }
+    {
+      name = "ignoreRepos";
+      description =
+        "A list of git repos to ignore. Repos are identified by their root directory.";
+      example = [ "/home/me/work/projects/foo" ];
+      type = types.listOf types.str;
+    }
+    {
+      name = "maxWidth";
+      description =
+        "Maximum width of the shell that the prompt may use, in percent. Setting this to 0 disables the shrinking subsystem.";
+      type = types.int;
+    }
+    {
+      name = "mode";
+      description = ''
+        The characters used to make separators between segments. (default "patched")'';
+      type = types.enum [ "patched" "compatible" "flat" ];
+    }
+    {
+      name = "modules";
+      description = ''
+        The list of modules to load (default "venv,user,host,ssh,cwd,perms,git,hg,jobs,exit,root")'';
+      type = types.listOf (types.enum moduleNames);
+    }
+    {
+      name = "newline";
+      description = "Show the prompt on a new line";
+      type = types.bool;
+    }
+    {
+      name = "numericExitCodes";
+      description = "Shows numeric exit codes for errors.";
+      type = types.bool;
+    }
+    {
+      name = "pathAliases";
+      description =
+        "One or more aliases from a path to a short name. An alias maps a path like foo/bar/baz to a short name like FBB. Use '~' for your home dir. You may need to escape this character to avoid shell substitution.";
+      example = {
+        "$GOPATH/src/github.com" = "@GOPATH-GH";
+        "~/work/projects/foo" = "@FOO";
+        "~/work/projects/bar" = "@BAR";
+      };
+      type = types.attrs;
```

Comment:
Best to avoid `types.attrs`, which merges poorly, and perhaps instead have `types.attrsOf`.

Reply:
fixed